opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/OpenSearchSortIndexScanRule.java

```diff
@@ -21,11 +21,11 @@ protected OpenSearchSortIndexScanRule(Config config) {
 
   @Override
   public void onMatch(RelOptRuleCall call) {
-    final LogicalSort sort = call.rel(0);
-    final CalciteLogicalIndexScan scan = call.rel(1);
+    final Sort sort = call.rel(0);
```

Comment:
@LantaoJin @qianheng-aws couple questions?
1. Is OpenSearchSortIndexScanRule intended to be a physical tweak rule for EnumerableRel?
2. Do we still need Logical plan optimization rule and CalciteLogicalIndexScan? Can all existing rules be expressed as physical tweak rules for EnumerableRel?

Reply:
EnumerableSort can generate from logical sort operator and logical join operator with physical SortMerge implementation. So I change the `OpenSearchSortIndexScanRule` rule from logical to physical layer. It equals keeping the `OpenSearchSortIndexScanRule` in logical layer and adding a new rule for EnumerableSort + SortMergeJoin in physical layer.

Almost pushdown rules can move to physical layer but I didn't see any benefit to do that. We need logical plan rules. some optimization should happen in logical layer instead of physical. For example a LogicalJoin operator may generate multiple physical Join operators. rule applying in logical should be more efficient.